packages/navie/src/services/vector-terms-service.ts

```diff
@@ -5,6 +5,7 @@ import { ChatOpenAI } from '@langchain/openai';
 
 import InteractionHistory, { VectorTermsInteractionEvent } from '../interaction-history';
 import trimFences from '../lib/trim-fences';
+import isEmpty from '../lib/is-empty';
```

Comment:
Overall I think it would be more effective to handle the case on the backend when the search terms are empty; and also to handle the empty search terms case within Navie and not not perform a context lookup if the vector terms are empty.

If the vector terms are empty, it's very unlikely that substituting the search with the user's raw search words is going to do any better. I haven't seen the LLM just fail to come up with any search terms as long as the user's question is at all reasonable (e.g. it isn't "HI").